### PR TITLE
SCRD-4708 Add a basic protractor test for day2

### PR DESCRIPTION
**Pull request by mserigithub**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,12 @@ To run the selenium tests:
 1. perform initial setup `npm run protractor-setup` (once, does not
    need to be done each time)
 2. start the app in another terminal `npm start`
-3. run the tests `npm run protractor`
+3. run the tests `npm run protractor-day0` (for running day0 Installer tests)
+<<<<<<< HEAD
+4. run the tests `npm run protractor-day2` (for running day2 CLM Admin Console tests)
+=======
+4. run the tests `npm run protractor-day20` (for running day2 CLM Admin Console tests)
```
Comment (jeremy-moffitt) [inline README.md]: this looks like a merge gone wrong....

**Comment by jeremy-moffitt** [comment]
Other than the readme containing bits of the merge, the code looks fine... but for some reason this doesnt run correctly on my system :-/ the chrome window pops up and just stays blank, guessing I'm doing something wrong

File: README.md (hunk 1 above)
Comment (mserigithub) [reply README.md]: I was trying to take out extra 0 from day20 and ended up with this state. Not sure how to fix this now.

**Comment by mserigithub** [comment]
> Other than the readme containing bits of the merge, the code looks fine... but for some reason this doesnt run correctly on my system :-/ the chrome window pops up and just stays blank, guessing I'm doing something wrong



> Other than the readme containing bits of the merge, the code looks fine... but for some reason this doesnt run correctly on my system :-/ the chrome window pops up and just stays blank, guessing I'm doing something wrong

Does localhost:2209 work in your local when you open manually?

File: README.md (hunk 1 above)
Comment (eapodaca) [reply README.md]: You should be able to checkout this branch and edit this file, commit those changes and push again.

Comment (mserigithub) [reply README.md]: Thanks. Fixed it.

**Comment by eapodaca** [comment]
The day2 specs work for me with a caveat. I had to adjust the `protractor-day2.conf.js` file `onPrepare` method with the following `browser.driver.manage().window().setSize(1920, 1000);` The window was launching with a size that was breaking one of the specs.

**Comment by mserigithub** [comment]
> The day2 specs work for me with a caveat. I had to adjust the `protractor-day2.conf.js` file `onPrepare` method with the following `browser.driver.manage().window().setSize(1920, 1000);` The window was launching with a size that was breaking one of the specs.

@eapodaca Does `browser.driver.manage().window().maximize();` work for you? I thought maximize can be a solution for any of the screen resolution.

**Comment by eapodaca** [comment]
@mserigithub maximize is probably fine.

**Comment by eapodaca** [comment]
I think the glob that is selecting specs in `protractor-day0.conf.js` needs to be updated. The way it is right now it tries to run the both the 'day0' and day2' specs.

**Comment by mserigithub** [comment]
> @mserigithub maximize is probably fine.

@eapodaca Ok. Changes made. Thanks.

**Comment by eapodaca** [comment]
@mserigithub Sorry we are asking for so many changes. Can we simplify the structure of the test directory? suggestion is we add a `day0` and `day2` directory `test` and keep the specs for the relevant mode in those directories, with directories under them to organize by page as desired etc. Then update the globs to be more generic `./tests/day0/**/*.spec.js` and `./tests/day2/**/*.spec.js`

**Comment by mserigithub** [comment]
> @mserigithub Sorry we are asking for so many changes. Can we simplify the structure of the test directory? suggestion is we add a `day0` and `day2` directory `test` and keep the specs for the relevant mode in those directories, with directories under them to organize by page as desired etc. Then update the globs to be more generic `./tests/day0/**/*.spec.js` and `./tests/day2/**/*.spec.js`

@eapodaca Absolutely. That was my plan to split the test cases based on the app and it is the reason, I created two separate config files. This review is very basic for day2 UI and I have many changing coming in next few days. Do you mind I take care of those changes in coming reviews?

**Comment by mserigithub** [comment]
> with the earlier version I could launch the tests in firefox but not chrome... now in FF I get an error message (though it does seem to try to navigate to Day0) `Failed: [Exception... "The URI scheme corresponds to an unknown protocol handler" nsresult: "0x804b0012 (NS_ERROR_UNKNOWN_PROTOCOL)" location: "JS frame :: chrome://marionette/content/listener.js :: get/< :: line 1053" data: no] Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53' System info: host: 'linux-qedk', ip: 'Unknown', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.159-73-default', java.version: '1.8.0_181' Driver info: driver.version: unknown`

@jeremy-moffitt It could be a setup issue. I hope you have already run `npm run protractor-setup` and changed browserName: 'firefox' in conf file. Next thing you could check is FF version. I am using 60.3.0esr (64-bit) and it works fine for me. I am on leap 42.3.

**Comment by mserigithub** [comment]
> > with the earlier version I could launch the tests in firefox but not chrome... now in FF I get an error message (though it does seem to try to navigate to Day0) `Failed: [Exception... "The URI scheme corresponds to an unknown protocol handler" nsresult: "0x804b0012 (NS_ERROR_UNKNOWN_PROTOCOL)" location: "JS frame :: chrome://marionette/content/listener.js :: get/< :: line 1053" data: no] Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53' System info: host: 'linux-qedk', ip: 'Unknown', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.159-73-default', java.version: '1.8.0_181' Driver info: driver.version: unknown`
> 
> @jeremy-moffitt It could be a setup issue. I hope you have already run `npm run protractor-setup` and changed browserName: 'firefox' in conf file. Next thing you could check is FF version. I am using 60.3.0esr (64-bit) and it works fine for me. I am on leap 42.3.

@jeremy-moffitt I have put a new patch. Could you please test with lastest code again and see if it resolved this issue.

**Comment by jeremy-moffitt** [comment]
Hmm I;m also on 60.3.0esr (64-bit)  with Leap 42.3... still the same problem... not sure whats different, I'll google around a bit

**Comment by jeremy-moffitt** [comment]
I got the first test to succeed by changing:
`browser.get('localhost:2209?installreset=true');`
to
`browser.get('http://localhost:2209?installreset=true');`
checking the later ones next

**Comment by jeremy-moffitt** [comment]
Got it working with the following changes (firefox appears to return rgb values rather than rgba):
```   beforeAll(function() {
-    browser.get('localhost:2209?installreset=true');
+    browser.get('http://localhost:2209?installreset=true');
     browser.sleep(1000);
   });
 
@@ -28,7 +28,7 @@ describe('basic sanity tests', function() {
     var stateLineWrapperContainer = element(by.css('.wizard-progress-container'));
     var firstIndicator = stateLineWrapperContainer.all(by.css('.progress')).first();
 
-    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
+    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
   });
 
   it('advances to the next page and updates the indicator', function() {
@@ -37,13 +37,13 @@ describe('basic sanity tests', function() {
     var lastIndicator = stateLineWrapperContainer.all(by.css('.progress')).last();
 
     //the first indicator is the correct "in-progress" color
-    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
+    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
 
     var nextButton = element(by.xpath('.//Button[.="Next"]'));
     nextButton.click();
     //the first indicator has updated to the "complete" color
-    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
+    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
     //the last indicator has a "notdone" color
-    expect(lastIndicator.getCssValue('background-color')).toEqual('rgba(106, 114, 118, 1)');//the rgba value of #6a7276
+    expect(lastIndicator.getCssValue('background-color')).toContain('106, 114, 118');//the rgba value of #6a7276
```

**Comment by jeremy-moffitt** [comment]
did not test in chrome with those changes since I cant seem to get it to run in Chrome

**Comment by GarySmith** [comment]
When I try to `npm run protractor-day0` (after `npm run protractor-setup`) I just get 
```
npm ERR! code ELIFECYCLE
npm ERR! errno 100
npm ERR! ardana-installer-ui@0.1.0 protractor-day0: `protractor protractor-day0.conf.js`
npm ERR! Exit status 100
npm ERR!
npm ERR! Failed at the ardana-installer-ui@0.1.0 protractor-day0 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```
with no meaningful info in the log. Am I doing something wrong?

**Comment by mserigithub** [comment]
> I got the first test to succeed by changing:
> `browser.get('localhost:2209?installreset=true');`
> to
> `browser.get('http://localhost:2209?installreset=true');`
> checking the later ones next

@jeremy-moffitt Thanks for catching. I fixed in the latest patch.

**Comment by mserigithub** [comment]
> Got it working with the following changes (firefox appears to return rgb values rather than rgba):
> 
> ```
> -    browser.get('localhost:2209?installreset=true');
> +    browser.get('http://localhost:2209?installreset=true');
>      browser.sleep(1000);
>    });
>  
> @@ -28,7 +28,7 @@ describe('basic sanity tests', function() {
>      var stateLineWrapperContainer = element(by.css('.wizard-progress-container'));
>      var firstIndicator = stateLineWrapperContainer.all(by.css('.progress')).first();
>  
> -    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
> +    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
>    });
>  
>    it('advances to the next page and updates the indicator', function() {
> @@ -37,13 +37,13 @@ describe('basic sanity tests', function() {
>      var lastIndicator = stateLineWrapperContainer.all(by.css('.progress')).last();
>  
>      //the first indicator is the correct "in-progress" color
> -    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
> +    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
>  
>      var nextButton = element(by.xpath('.//Button[.="Next"]'));
>      nextButton.click();
>      //the first indicator has updated to the "complete" color
> -    expect(firstIndicator.getCssValue('background-color')).toEqual('rgba(0, 192, 129, 1)');//the rgba value of #00C081
> +    expect(firstIndicator.getCssValue('background-color')).toContain('0, 192, 129');//the rgba value of #00C081
>      //the last indicator has a "notdone" color
> -    expect(lastIndicator.getCssValue('background-color')).toEqual('rgba(106, 114, 118, 1)');//the rgba value of #6a7276
> +    expect(lastIndicator.getCssValue('background-color')).toContain('106, 114, 118');//the rgba value of #6a7276
> ```

@jeremy-moffitt Thanks again. I fixed in the latest patch.

**Comment by mserigithub** [comment]
> When I try to `npm run protractor-day0` (after `npm run protractor-setup`) I just get
> 
> ```
> npm ERR! code ELIFECYCLE
> npm ERR! errno 100
> npm ERR! ardana-installer-ui@0.1.0 protractor-day0: `protractor protractor-day0.conf.js`
> npm ERR! Exit status 100
> npm ERR!
> npm ERR! Failed at the ardana-installer-ui@0.1.0 protractor-day0 script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> ```
> with no meaningful info in the log. Am I doing something wrong?

@GarySmith Sorry. Could you please try with latest patch? I believe you have run npm install before running the protractor test. And application is accessible at the specified port.

**Pull request merged**